tests/unit/test_jobs.py

```diff
@@ -0,0 +1,33 @@
+from catkin_tools.jobs.utils import merge_envs
+
+
+def test_merge_envs_basic():
+    job_env = { 'PATH': '/usr/local/bin:/usr/bin', 'FOO': 'foo' }
+
+    merge_envs(job_env, [
+        { 'PATH': '/usr/local/bin:/bar/baz/bin' },
+        { 'BAR': 'bar' } ])
+
+    # Validate that the known path was not moved from the existing order, and the unfamiliar
+    # path was correctly prepended.
+    assert job_env['PATH'] == '/bar/baz/bin:/usr/local/bin:/usr/bin'
```

Comment:
The general merging case is much harder that this one— here we benefit from a bunch of assumptions, chiefly that all the envs being merged share a common underlay, and that since there are expected to be no duplicate files across the set of workspace resultspaces, the order they end up in in the overlay doesn't actually matter, as long as they're all present in the final list of paths, and all ahead of the underlay paths.